### ISSUE TITLE
test_compat fails on Windows Python 3

### ISSUE THREAD

**Issue by twisted-trac** [twisted/twisted#8757]
|[<img alt="rodrigc's avatar" src="https://avatars.githubusercontent.com/u/1895943?s=50" width="50" height="50">](https://github.com/rodrigc)| @rodrigc reported|
|-|-|
|Trac ID|trac#8757|
|Type|enhancement|
|Created|2016-08-14 09:45:32Z|
|Branch|https://github.com/twisted/twisted/tree/8757-rodrigc-compat-py3|

On Windows with Python 3.5, test_compat fails:

```
[FAIL]
Traceback (most recent call last):
  File "c:\projects\twisted\build\py35-coverage-windows\lib\site-packages\twisted\test\test_compat.py", line 771, in test_alwaysBytes
    self.assertEqual(list(types), [bytes])
  File "c:\projects\twisted\build\py35-coverage-windows\lib\site-packages\twisted\trial\_synctest.py", line 425, in assertEqual
    super(_Assertions, self).assertEqual(first, second, msg)
  File "C:\PYTHON35-x64\Lib\unittest\case.py", line 820, in assertEqual
    assertion_func(first, second, msg=msg)
  File "C:\PYTHON35-x64\Lib\unittest\case.py", line 1018, in assertListEqual
    self.assertSequenceEqual(list1, list2, msg, seq_type=list)
  File "C:\PYTHON35-x64\Lib\unittest\case.py", line 1000, in assertSequenceEqual
    self.fail(msg)
twisted.trial.unittest.FailTest: Lists differ: [<class 'str'>] != [<class 'bytes'>]
First differing element 0:
<class 'str'>
<class 'bytes'>
- [<class 'str'>]
?           --
+ [<class 'bytes'>]
?          ++++
twisted.test.test_compat.BytesEnvironTests.test_alwaysBytes
```

<details><summary>Searchable metadata</summary>

```
trac-id__8757 8757
type__enhancement enhancement
reporter__rodrigc rodrigc
priority__normal normal
milestone__Python_3_x Python-3.x
branch__8757_rodrigc_compat_py3 8757-rodrigc-compat-py3
branch_author__ 
status__closed closed
resolution__fixed fixed
component__core core
keywords__None None
time__1471167932796641 1471167932796641
changetime__1471560802042098 1471560802042098
version__None None
owner__Craig_Rodrigues__rodrigc_____ Craig Rodrigues <rodrigc@...>

```
</details>

**Comment by twisted-trac** [comment]
|[<img alt="rodrigc's avatar" src="https://avatars.githubusercontent.com/u/1895943?s=50" width="50" height="50">](https://github.com/rodrigc)<a name="note_1"></a>|@rodrigc commented|
|-|-|

https://github.com/twisted/twisted/pull/496

**Comment by twisted-trac** [comment]
|[<img alt="acabhishek942's avatar" src="https://avatars.githubusercontent.com/u/7778186?s=50" width="50" height="50">](https://github.com/acabhishek942)<a name="note_2"></a>|@acabhishek942 commented|
|-|-|

changes looks good. Please merge

**Comment by twisted-trac** [comment]
|[<img alt="rodrigc's avatar" src="https://avatars.githubusercontent.com/u/1895943?s=50" width="50" height="50">](https://github.com/rodrigc)<a name="note_3"></a>|@rodrigc set owner to @rodrigc<br>@rodrigc set status to `closed`|
|-|-|

In changeset 36adf49661456829c085e5e35fc4b0fe3222d3ae
```
#!CommitTicketReference repository="" revision="36adf49661456829c085e5e35fc4b0fe3222d3ae"
Merge pull request #496 from twisted/8757-rodrigc-compat-py3

Author: hawkowl, rodrigc
Reviewer: acabhishek942
Fixes: #8757

Fix test_compat on Windows with Python 3
```
